File: Python/6 kyu/0003.py
# Kata: Non-even substrings
# Link: https://www.codewars.com/kata/59da47fa27ee00a8b90000b4

# Given a string of integers, return the number of odd-numbered substrings that
# can be formed.

# For example, in the case of "1341", they are 1, 1, 3, 13, 41, 341, 1341, a
# total of 7 numbers.

# solve("1341") = 7. See test cases for more examples.

# Good luck!


def solve(s) -> int:
    even_digits = list(map(is_even, list(s)))
    count: int = 0
    for i in range(len(s)):
        j: int = i
        while j < len(s):
            if not even_digits[j]:
                count += 1
            j += 1
    return count


def is_even(c: str):
    return int(c) % 2 == 0

File: JavaScript/6 kyu/0003.js
// Kata: IQ Test
// Link: https://www.codewars.com/kata/552c028c030765286c00007d

// Bob is preparing to pass IQ test. The most frequent task in this test is to find out which one of the given numbers differs from the others. Bob observed that one number usually differs from the others in evenness. Help Bob — to check his answers, he needs a program that among the given numbers finds one that is different in evenness, and return a position of this number.

// ! Keep in mind that your task is to help Bob solve a real IQ test, which means indexes of the elements start from 1 (not 0)

// Examples:
//    iqTest("2 4 7 8 10") => 3 // Third number is odd, while the rest of the numbers are even

//    iqTest("1 2 1 1") => 2 // Second number is even, while the rest of the numbers are odd

function iqTest(numbersString) {
  const numbers = numbersString.split(' ');
  let odd = [],
    even = [];
  for (let i = 0; i < numbers.length; ++i) {
    if (numbers[i] % 2 === 0) {
      even.push(i + 1);
    } else {
      odd.push(i + 1);
    }
    if (even.length === 1 && odd.length > 1) return even[0];
    if (odd.length === 1 && even.length > 1) return odd[0];
  }
}
